src/GUI/RevEng.Core.50/CommentCSharpEntityTypeGenerator.cs

```diff
@@ -274,7 +285,22 @@ protected virtual void GenerateProperties([NotNull] IEntityType entityType)
                     GeneratePropertyDataAnnotations(property);
                 }
 
-                _sb.AppendLine($"public {_code.Reference(property.ClrType)} {property.Name} {{ get; set; }}");
+                string nullableAnnotation = string.Empty;
+                string defaultAnnotation = string.Empty;
+
+                if (_nullableReferences && !property.ClrType.IsValueType)
+                {
+                    if (property.IsColumnNullable())
+                    {
+                        nullableAnnotation = "?";
+                    }
+                    else
+                    {
+                        defaultAnnotation = $" = default!;";
```

Comment:
This is indeed one way to get around the property initialization problem, but the more robust/recommended way IMHO should be with a nullable backing field and a non-nullable property whose getter throws on null ([see the docs](https://docs.microsoft.com/en-us/ef/core/miscellaneous/nullable-reference-types#non-nullable-properties-and-initialization)).

That's definitely more elaborate/complicated than this though...

Reply:
Thanks, @roji I really appreciate you taking time to have a look!
For now, I will just go with this simpler approach 😄